lib/tasks/TaskUtil.js

```diff
@@ -0,0 +1,154 @@
+	 * @param {string} tag Name of the tag
+	 * @returns {string|boolean|integer|undefined} Tag value for the given resource.
+	 * 										<code>undefined</code> if no value is available
+	 * @public
+	 */
+	getTag(resource, tag) {
+		return this._projectBuildContext.getResourceTagCollection().getTag(resource, tag);
+	}
+
+	/**
+	 * Clears the value of a tag stored for the given resource's path.
+	 * It's like the tag was never set for that resource.
+	 *
+	 * </br></br>
+	 * This method is only available to custom task extensions defining
+	 * <b>Specification Version 2.2 and above</b>.
+	 *
+	 * @param {module:@ui5/fs.Resource} resource Resource the tag should be cleared for
+	 * @param {string} tag Tag
+	 * @public
+	 */
+	clearTag(resource, tag) {
+		return this._projectBuildContext.getResourceTagCollection().clearTag(resource, tag);
+	}
+
+	/**
+	 * Check whether the project currently being build is the root project.
+	 *
+	 * </br></br>
+	 * This method is only available to custom task extensions defining
+	 * <b>Specification Version 2.2 and above</b>.
+	 *
+	 * @returns {boolean} <code>true</code> if the currently built project is the root project
+	 * @public
+	 */
+	isRootProject() {
+		return this._projectBuildContext.isRootProject();
+	}
+
+	/**
+	 * Register a function that must be executed once the build is finished. This can be used to, for example,
+	 * cleanup files temporarily created on the file system. If the callback returns a Promise, it will be waited for.
+	 * It will also be executed in cases where the build has failed or has been aborted.
+	 *
+	 * </br></br>
+	 * This method is only available to custom task extensions defining
+	 * <b>Specification Version 2.2 and above</b>.
+	 *
+	 * @param {Function} callback Callback to register. If it returns a Promise, it will be waited for
+	 * @public
+	 */
+	registerCleanupTask(callback) {
+		return this._projectBuildContext.registerCleanupTask(callback);
+	}
+
+	/**
+	 * Get an interface to an instance of this class that only provides those functions
+	 * that are supported by the given custom task extension Specification Version.
+	 *
+	 * @param {string} specVersion Specification Version of custom task extension
+	 * @returns {object} An object with bound instance methods supported by the given Specification Version
+	 */
+	getInterface(specVersion) {
+		if (["0.1", "1.0", "1.1", "2.0", "2.1"].includes(specVersion)) {
+			return undefined;
+		}
+
+		const baseInterface = {
+			STANDARD_TAGS: this.STANDARD_TAGS,
+			setTag: this.setTag.bind(this),
+			clearTag: this.clearTag.bind(this),
+			getTag: this.getTag.bind(this),
+			isRootProject: this.isRootProject.bind(this),
+			registerCleanupTask: this.registerCleanupTask.bind(this)
+		};
+		switch (specVersion) {
+		case "2.2":
+			return baseInterface;
+		default:
+			throw new Error(`TaskUtil: Unknown or unsupported Specification Version ${specVersion}`);
```

Comment:
```suggestion
			throw new Error(`TaskUtil: Unknown or unsupported specification version ${specVersion}`);
```

Reply:
This will be visible to the user as `TaskUtil: Unknown or unsupported Specification Version 4.0`. Should we still use lower case "Specification Version"?

Reply:
You're right, in this case we should leave it upper case. Thanks.